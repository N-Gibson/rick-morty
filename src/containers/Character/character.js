import React from 'react';
import './character.scss';
import { PropTypes } from 'prop-types';

const Character = ({ name, status, species, gender, image, location, origin }) => {
  return (
    <div className='character-card'>
      <h2>{name}</h2>
      <img src={image}></img>
      <div>
        <p>Species: {species}</p>
        <p>Status: {status}</p>
        <p>Location: {location}</p>
        <p>Gender: {gender}</p>
        <p>Origin: {origin}</p>
      </div>
    </div>
  )
}

export default Character

Character.propTypes = {
  name: PropTypes.string,
  status: PropTypes.string,
  species: PropTypes.string,
  gender: PropTypes.string,
  image: PropTypes.string,
  location: PropTypes.string,
  origin: PropTypes.string
}