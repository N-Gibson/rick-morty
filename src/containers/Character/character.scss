.character-card {
  background-image: url('https://media.giphy.com/media/hsITL773u90X6KprRR/giphy.gif');
  background-size: auto;
  border-radius: 5px;
    &:hover {
      box-shadow: 0px 0px 10px 2px rgb(140, 244, 46);
    }
}

img {
  width: 100%;
}

h2 {
  color: #fff;
  padding: 0px 10px;
}

p {
  color: #fff;
  padding: 0px 10px;
  text-align: center
}