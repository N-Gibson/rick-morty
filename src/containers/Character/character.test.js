import React from 'react';
import Character from '../Character/character';
import { shallow } from 'enzyme';

describe('Character', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<Character />);
    expect(wrapper).toMatchSnapshot();
  })
});