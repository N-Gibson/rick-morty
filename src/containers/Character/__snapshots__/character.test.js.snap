// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character should match snapshot 1`] = `ShallowWrapper {}`;
